### PR TITLE
#160577603 Configure app for Heroku deployment

### PR DESCRIPTION
**Pull request by mutaimwiti**
### What does this PR do?
Setup and deploy base application to Heroku. This is accomplished by adding `Heroku` `app.json` and `Procfile` to the project root. The `app.json` specifies how apps are created by the project pipeline. The `Procfile` specifies how the project will be run. `gunicorn` WSGI server is used.

### Description of the task to be completed?
GIVEN application URLs
WHEN user navigates to provided URLs in browser
THEN user should access the live application

GIVEN a developer raises a Pull Request
WHEN a review app is automatically created
THEN review app should mirror the Pull Request being raised

`staging` and `production` environments are to be setup and URLs should follow the naming pattern below:
- `<repo-name>-staging.herokuapp.com` for `staging`
- `<repo-name>.herokuapp.com` for `production`

### How this should be manually tested?
This PR can be manually tested by raising a pull request. This triggers auto-deployment of a review app for the PR. When code is pushed to the `develop` branch the `staging` app is automatically deployed at [https://ah-code-blooded-staging.herokuapp.com](https://ah-code-blooded-staging.herokuapp.com). When code is pushed to the `master` branch the `production` app is automatically deployed at [https://ah-code-blooded.herokuapp.com](https://ah-code-blooded.herokuapp.com). 

### What are the relevant pivotal tracker stories?
[#160577603](https://www.pivotaltracker.com/story/show/160577603)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ django-cors-middleware==1.3.1
 django-extensions==2.1.2
 djangorestframework==3.8.2
 djangorestframework-jwt==1.11.0
+gunicorn==19.9.0
```
Comment (gitaumoses4) [inline requirements.txt:7]: Could you add this part in the commit message too?

**Review by gitaumoses4 — APPROVED** [review]
Follows conventions to the letter. Perfect.

File: requirements.txt (hunk 1 above)
Comment (mutaimwiti) [reply requirements.txt:7]: > Could you add this part in the commit message too?

Yes, I was trying to keep it as concise as possible.

Comment (gitaumoses4) [reply requirements.txt:7]: Ah. Good good. Makes sense. Nice.

**Comment by charisschomba** [comment]
Nice description on the Pull Request.

File: app.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "ah-code-blooded",
+  "scripts": {},
+  "env": {
+    "DISABLE_COLLECTSTATIC": {
```
Comment (esirK) [inline app.json:5]: Why `DISABLE_COLLECTSTATIC` ??

**Comment by esirK** [comment]
Good job. You may need to add `ah-code-blooded-test.herokuapp.com` on the list of allowed hosts.

**Comment by bevkololi** [comment]
You've set a good example on the PR description.

File: app.json (hunk 2 above)
Comment (mutaimwiti) [reply app.json:5]: I'm disabled it since the API is not serving static content.

Comment (mutaimwiti) [reply app.json:5]: After some consideration, I think it needs to serve static content for the browsable API.

**Pull request merged**